shell/platform/darwin/ios/framework/Source/FlutterPluginAppLifeCycleDelegate.mm

```diff
@@ -143,7 +144,10 @@ - (void)handleDidEnterBackground:(NSNotification*)notification {
   _debugBackgroundTask = [application
       beginBackgroundTaskWithName:@"Flutter debug task"
                 expirationHandler:^{
-                  [application endBackgroundTask:_debugBackgroundTask];
+                  if (_debugBackgroundTask != UIBackgroundTaskInvalid) {
+                    [application endBackgroundTask:_debugBackgroundTask];
```

Comment:
Does this need to be done in dealloc as well (add-to-app where this delegate may actually be deallocated)?

Reply:
I don't think so.  The background task has a retain on the FlutterPluginAppLifeCycleDelegate so it can't be dealloced while it is running.